messaging/index.d.ts

```diff
@@ -1,3 +1,18 @@
 export function follow<T, U>(previous: T, classOfNext: new (...args: any[]) => U, additionalFields?: Partial<U>): U;
 
+export type Metadata = {
+  follow: Function;
+  globalPosition: number;
+  position: number;
+  stream: string;
+  time: string;
+  causationMessageStreamName?: string;
+  causationMessagePosition?: number;
+  causationMessageGlobalPosition?: number;
+  correlationStreamName?: string;
+  replyStreamName?: string;
+} & {
```

Comment:
Can you not do `& object` here as they are pretty much equivalent?

Reply:
Just tested this out - they don't work the same.  The `[prop: string]: any;` allows any arbitrary string-keyed property to exist on the `metadata` object, allowing for `metadata.asfdslkjfs`.

Updating this to `& object` has TS errors saying `Property 'asfdslkjfs' does not exist on type Metadata`.

That said, I'm wishing I didn't have to use `any` on the right side, since Functions aren't valid JSON, but deeply-nested objects / Arrays are.